src/ytdl_sub/cli/download_args_parser.py

```diff
@@ -116,10 +152,48 @@ def to_subscription_dict(self) -> Dict:
             if not self._is_argument_name(arg=argument_name):
                 raise self._argument_exception
 
-            argument_dict = self._argument_name_and_value_to_dict(
-                arg_name=argument_name, arg_value=argument_value
-            )
-            mergedeep.merge(subscription_dict, argument_dict, strategy=mergedeep.Strategy.REPLACE)
+            prefix, delim, last = argument_name.rpartition(".")
+            list_index = self._get_list_index(last) if delim and last else None
+
+            if list_index is None:
+                new_arguments.append(argument_name)
+                new_arguments.append(argument_value)
+                continue
+
+            if prefix not in list_arguments:
+                list_arguments[prefix] = ([], [])
+
+            list_arguments[prefix][0].append(list_index - 1)
+            list_arguments[prefix][1].append(argument_value)
+
+        for name, value in list_arguments.items():
+            n = self._find_largest_consecutive(indices=(value[0]).copy()) + 1
+
+            if n == 0:
+                raise InvalidDlArguments("Incomplete list")
+
+            list_value = [""] * n
+
+            for i in range(len(value[0])):
+                if value[0][i] < n:
+                    list_value[value[0][i]] = value[1][i]
+
+            new_arguments.append(name)
+            new_arguments.append(list_value)
+
+        subscription_dict = {}
+        arguments = new_arguments
+
+        try:
+            for idx in range(0, len(arguments), 2):
+                argument_name, argument_value = arguments[idx], arguments[idx + 1]
+
+                argument_dict = self._argument_name_and_value_to_dict(
+                    arg_name=argument_name, arg_value=argument_value
+                )
+                mergedeep.merge(subscription_dict, argument_dict, strategy=mergedeep.Strategy.TYPESAFE_REPLACE)
+        except (TypeError):
+            raise InvalidDlArguments("Cannot specify a single variable to be two different types")
```

Comment:
Lets include the problematic argument name in the error message so its more verbose.
`f"Invalid dl argument {argument_name}: Cannot specify an argument to be two different types"`